src/query/users/src/role_cache_mgr.rs

```diff
@@ -111,6 +118,63 @@ impl RoleCacheManager {
         Ok(cached_roles.roles.get(role).cloned())
     }
 
+    #[async_backtrace::framed]
+    pub async fn find_object_owner(
```

Comment:
this method seems not quite related with `role_cache`, any other suggested place to put this method?

Reply:
UserApiProvider seems to be some api interface provider which may also not make sense to provide this meta service logic on it.

Reply:
I think the root issue is that we do not want to directly put an api client in Session struct, as it may get leaked to introduce unwanted trouble

Reply:
I see, I guess a seperated `PrivilegeManager` may be a better code layout than floating all the things into `Session`. Can you leave a TODO on top of this function? I'd like to try adding a PrivilegeManager later after this PR get merged, to collect the fragmented privilege logics into one place.